src/main/java/BottleOpener/Event.java

```diff
@@ -0,0 +1,161 @@
+package BottleOpener;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
+
+/**
+ * Represents an event with a start and end time.
+ * <p>
+ * An BottleOpener.Event task includes a description, a status indicating whether it is completed,
+ * and a start and end time represented as {@link LocalDateTime}.
+ * </p>
+ */
+public class Event extends Task {
+    private static final DateTimeFormatter FORMATTER;
+
+    static {
+        FORMATTER = new DateTimeFormatterBuilder()
+                .appendOptional(DateTimeFormatter.ofPattern("dd MMM yyyy HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("dd MMM yyyy HH:mm"))
+                .appendOptional(DateTimeFormatter.ofPattern("dd MMM yyyy"))
+                .appendOptional(DateTimeFormatter.ofPattern("dd/M/yyyy HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("dd/M/yyyy HH:mm"))
+                .appendOptional(DateTimeFormatter.ofPattern("dd/M/yyyy"))
+                .appendOptional(DateTimeFormatter.ofPattern("d/M/yyyy"))
+                .appendOptional(DateTimeFormatter.ofPattern("d/M/yyyy HH:mm"))
+                .appendOptional(DateTimeFormatter.ofPattern("d/M/yyyy HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("MM/dd/yyyy HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("MM/dd/yyyy HH:mm"))
+                .appendOptional(DateTimeFormatter.ofPattern("MM/dd/yyyy"))
+                .appendOptional(DateTimeFormatter.ofPattern("MM-dd-yyyy HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("MM-dd-yyyy HH:mm"))
+                .appendOptional(DateTimeFormatter.ofPattern("MM-dd-yyyy"))
+                .parseDefaulting(ChronoField.HOUR_OF_DAY, 0)
+                .parseDefaulting(ChronoField.MINUTE_OF_HOUR, 0)
+                .toFormatter();
+    }
```

Comment:
Duplicated code, maybe can abstract them away? or just have one of the class import it from the other class